### PR TITLE
Update Remove-WinADDuplicateObject.ps1

### PR DESCRIPTION
**Pull request by Manukumarkm**
Added a check to handle ProtectedFromAccidentalDeletion true scenarios.

### CODE REVIEW

File: Public/Remove-WinADDuplicateObject.ps1 [hunk 1]
```diff
@@ -24,6 +24,9 @@
     $Count = 0
     $DuplicateObjects = Get-WinADDuplicateObject @getWinADDuplicateObjectSplat
     foreach ($Duplicate in $DuplicateObjects | Select-Object -First $LimitProcessing) {
+        If($Duplicate.ProtectedFromAccidentalDeletion -eq "True"){
+            Get-ADObject -Filter {ObjectGUID -eq "$($Duplicate.ObjectGUID)"} -ErrorAction Stop | Set-ADObject -ProtectedFromAccidentalDeletion $false
```
Comment (PrzemyslawKlys) [inline Public/Remove-WinADDuplicateObject.ps1:28]: You need Error Action on Set-AdObject as well, and you need to have try/catch logic for that and decide what happens next. My guess would be - if this happens you need to continue the loop for next object, and not let Remove part to happen.

File: Public/Remove-WinADDuplicateObject.ps1 [hunk 2]
```diff
@@ -24,6 +24,9 @@
     $Count = 0
     $DuplicateObjects = Get-WinADDuplicateObject @getWinADDuplicateObjectSplat
     foreach ($Duplicate in $DuplicateObjects | Select-Object -First $LimitProcessing) {
+        If($Duplicate.ProtectedFromAccidentalDeletion -eq "True"){
```
Comment (PrzemyslawKlys) [inline Public/Remove-WinADDuplicateObject.ps1:27]: It should be $true - it's not a string, it's a bool as far as I know

**Review by PrzemyslawKlys — COMMENTED** [review]
Thank you, please address two issues

File: Public/Remove-WinADDuplicateObject.ps1 (hunk 1 above)
Comment (Manukumarkm) [reply Public/Remove-WinADDuplicateObject.ps1]: Added error action for Set-AdObject, also added try catch to handle error scenarios.

File: Public/Remove-WinADDuplicateObject.ps1 (hunk 2 above)
Comment (Manukumarkm) [reply Public/Remove-WinADDuplicateObject.ps1]: Modified it to $true

File: Public/Remove-WinADDuplicateObject.ps1 [hunk 3]
```diff
@@ -24,8 +24,14 @@
     $Count = 0
     $DuplicateObjects = Get-WinADDuplicateObject @getWinADDuplicateObjectSplat
     foreach ($Duplicate in $DuplicateObjects | Select-Object -First $LimitProcessing) {
-        If($Duplicate.ProtectedFromAccidentalDeletion -eq "True"){
-            Get-ADObject -Filter {ObjectGUID -eq "$($Duplicate.ObjectGUID)"} -ErrorAction Stop | Set-ADObject -ProtectedFromAccidentalDeletion $false
+        If($Duplicate.ProtectedFromAccidentalDeletion -eq $true){
+            Try{
+                 Get-ADObject -Filter {ObjectGUID -eq "$($Duplicate.ObjectGUID)"} -ErrorAction Stop | Set-ADObject -ProtectedFromAccidentalDeletion $false -ErrorAction Stop
```
Comment (PrzemyslawKlys) [inline Public/Remove-WinADDuplicateObject.ps1:29]: Now that I think about it, Get-ADObject is not necessary. ObjectGUID is available as part of Duplicate.ObjectGUID, so is protected from accidental deletion so you can use Set-ADObject only.

**Comment by PrzemyslawKlys** [comment]
Thank you @Manukumarkm :-)

File: Public/Remove-WinADDuplicateObject.ps1 (hunk 3 above)
Comment (Manukumarkm) [reply Public/Remove-WinADDuplicateObject.ps1]: Yes, you are correct. removed Get-AdObject

**Pull request merged**